cartography/intel/kubernetes/eks.py

```diff
@@ -1,25 +1,276 @@
 import neo4j
 import yaml
+from arn import Arn
 from kubernetes.client.models import V1ConfigMap
 
 from cartography.client.core.tx import load
 from cartography.graph.job import GraphJob
+from cartography.intel.kubernetes.rbac import get_cluster_role_bindings
+from cartography.intel.kubernetes.rbac import get_role_bindings
+from cartography.intel.kubernetes.rbac import transform_users
 from cartography.intel.kubernetes.util import K8sClient
 from cartography.models.kubernetes.groups import KubernetesGroupSchema
 from cartography.models.kubernetes.oidc import KubernetesOIDCProviderSchema
 from cartography.models.kubernetes.users import KubernetesUserSchema
+from cartography.util import aws_handle_regions
 from cartography.util import timeit
 
 logger = logging.getLogger(__name__)
 
 
+class ParsedAuthMappings(TypedDict):
+    roles: List[Dict[str, Any]]
+    users: List[Dict[str, Any]]
+    accounts: List[str]
+    templated_roles: List[Dict[str, Any]]
+    templated_users: List[Dict[str, Any]]
+
+
+def process_templated_string(template_string: str, arn: str) -> str:
+    """
+    Process templated string by replacing template variables with actual values.
+
+    Currently supports:
+    - {{AccountID}}: Replaced with account ID from the provided ARN
+    - {{SessionNameRaw}}: Cannot be resolved at ConfigMap parse time, requires regex matching
+    - {{SessionName}}: Cannot be resolved at ConfigMap parse time, requires regex matching
+    """
+    if not template_string or not arn:
+        return template_string
+
+    processed_string = template_string
+
+    # Handle {{AccountID}} template first
+    if "{{AccountID}}" in processed_string:
+        try:
+            parsed_arn = Arn(arn)
+            account_id = parsed_arn.account
+            processed_string = processed_string.replace("{{AccountID}}", account_id)
+            logger.debug(
+                f"Replaced {{{{AccountID}}}} with {account_id} in string: {template_string} -> {processed_string}"
+            )
+        except Exception:
+            logger.warning(
+                f"Failed to parse account ID from ARN {arn} for templated string {template_string}"
+            )
+            return template_string  # Return original if we can't parse
+
+    # Check for SessionNameRaw templates - these cannot be resolved at parse time
+    # Return the partially processed string (with AccountID resolved) for regex matching
+    if (
+        "{{SessionNameRaw}}" in processed_string
+        or "{{SessionName}}" in processed_string
+    ):
+        logger.debug(
+            f"String contains session name template that requires regex matching: {processed_string}"
+        )
+        return processed_string  # Return partially processed string
+
+    return processed_string
+
+
+def template_to_regex(template_string: str) -> str:
+    """
+    Convert a templated string into a regular expression pattern.
+
+    Supports:
+    - {{AccountID}}: Matches 12-digit AWS account IDs
```

Comment:
We should not have to regex match on account IDs because we'll statically put these into the template. We'll always know the account ID from the ARN of the role in either mapUsers or mapRoles or from the list of account IDs in mapAccounts